internal/config/config.go

```diff
@@ -46,60 +49,76 @@ func (c *Config) DeepCopy() *Config {
 		ThriftProtocol:            c.ThriftProtocol,
 		ThriftTransport:           c.ThriftTransport,
 		ThriftProtocolVersion:     c.ThriftProtocolVersion,
 		ThriftDebugClientProtocol: c.ThriftDebugClientProtocol,
-		DriverName:                c.DriverName,
-		DriverVersion:             c.DriverVersion,
 	}
 }
 
 // UserConfig is the set of configurations exposed to users
 type UserConfig struct {
-	Protocol            string
-	Host                string // from databricks UI
-	Port                int    // from databricks UI
-	HTTPPath            string // from databricks UI
-	Catalog             string
-	Schema              string
-	AccessToken         string // from databricks UI
-	MaxRows             int    // TODO
-	QueryTimeoutSeconds int    // There are several timeouts that can be possibly configurable
-	UserAgentEntry      string
-	Location            *time.Location
-	SessionParams       map[string]string
+	Protocol       string
+	Host           string // from databricks UI
+	Port           int    // from databricks UI
+	HTTPPath       string // from databricks UI
+	Catalog        string
+	Schema         string
+	AccessToken    string        // from databricks UI
+	MaxRows        int           // max rows per page
+	QueryTimeout   time.Duration // Timeout passed to server for query processing
+	UserAgentEntry string
+	Location       *time.Location
+	SessionParams  map[string]string
 }
 
 func (ucfg UserConfig) DeepCopy() UserConfig {
-	sessionParams := make(map[string]string)
-	for k, v := range ucfg.SessionParams {
-		sessionParams[k] = v
+	var sessionParams map[string]string
+	if ucfg.SessionParams != nil {
+		sessionParams = make(map[string]string)
+		for k, v := range ucfg.SessionParams {
+			sessionParams[k] = v
+		}
 	}
+	var loccp *time.Location
+	if ucfg.Location != nil {
+		var err error
+		loccp, err = time.LoadLocation(ucfg.Location.String())
+		if err != nil {
+			logger.Warn().Msg("could not copy location")
+		}
+
+	}
+
 	return UserConfig{
-		Protocol:            ucfg.Protocol,
-		Host:                ucfg.Host,
-		Port:                ucfg.Port,
-		HTTPPath:            ucfg.HTTPPath,
-		Catalog:             ucfg.Catalog,
-		Schema:              ucfg.Schema,
-		AccessToken:         ucfg.AccessToken,
-		MaxRows:             ucfg.MaxRows,
-		QueryTimeoutSeconds: ucfg.QueryTimeoutSeconds,
-		UserAgentEntry:      ucfg.UserAgentEntry,
-		SessionParams:       sessionParams,
+		Protocol:       ucfg.Protocol,
+		Host:           ucfg.Host,
+		Port:           ucfg.Port,
+		HTTPPath:       ucfg.HTTPPath,
+		Catalog:        ucfg.Catalog,
+		Schema:         ucfg.Schema,
+		AccessToken:    ucfg.AccessToken,
+		MaxRows:        ucfg.MaxRows,
+		QueryTimeout:   ucfg.QueryTimeout,
+		UserAgentEntry: ucfg.UserAgentEntry,
+		Location:       loccp,
+		SessionParams:  sessionParams,
 	}
 }
 
-func (ucfg UserConfig) FillDefaults() UserConfig {
+func (ucfg UserConfig) WithDefaults() UserConfig {
 	if ucfg.MaxRows == 0 {
```

Comment:
Could maybe do `<= 0` in case someone tries to have a negative MaxRows

Reply:
ah, good one